src/lambda/commands/createNewSamApp.ts

```diff
@@ -352,21 +352,26 @@ async function validateSamCli(samCliValidator: SamCliValidator): Promise<void> {
 
 export async function getProjectUri(
     config: Pick<CreateNewSamAppWizardResponse, 'location' | 'name'>,
-    file: string
+    files: string[]
 ): Promise<vscode.Uri | undefined> {
-    const cfnTemplatePath = path.resolve(config.location.fsPath, config.name, file)
-    if (await fileExists(cfnTemplatePath)) {
-        return vscode.Uri.file(cfnTemplatePath)
-    } else {
-        vscode.window.showWarningMessage(
-            localize(
-                'AWS.samcli.initWizard.source.error.notFound',
-                'Project created successfully, but {0} file not found: {1}',
-                file,
-                cfnTemplatePath
-            )
-        )
+    let file: string
+    let cfnTemplatePath: string
+    for (const f of files) {
+         file = f
+         cfnTemplatePath = path.resolve(config.location.fsPath, config.name, file)
+        
+        if (await fileExists(cfnTemplatePath)) {
+            return vscode.Uri.file(cfnTemplatePath)
+        }
     }
+    vscode.window.showWarningMessage(
+        localize(
+            'AWS.samcli.initWizard.source.error.notFound',
+            'Project created successfully, but {0} file not found: {1}',
+            file!,
+            cfnTemplatePath!
```

Comment:
This is fragile. Instead of just assuming that `files` has items, we should assert that it does. Then these `!` suffixes won't be needed.

The warning was hinting at this. We want to assert the truth rather than just assume it.

First line of `getProjectUri()` should be:

```
if (files.length === 0) {
    throw Error('expected "files" parameter to have at least one item')
}
```

Reply:
I tried that previously, but I'm guessing it's not sophisticated enough to pick that up.  I've added the validation but had to leave in the `!` operator to alleviate the error.  This will at least eliminate the chance to pass null.